File: 백준/Silver/11728. 배열 합치기/배열 합치기.py
import sys

input = sys.stdin.readline
n, m = map(int, input().strip().split())
A = list(map(int, input().strip().split()))
B = list(map(int, input().strip().split()))

answer = []
i = j = 0

while i < n and j < m:
    if A[i] < B[j]:
        answer.append(A[i])
        i += 1
    else:
        answer.append(B[j])
        j += 1

if i < n:
    answer.extend(A[i:])
else:
    answer.extend(B[j:])

print(*answer)

File: 백준/Silver/11728. 배열 합치기/배열 합치기.js
const fs = require("fs");

const filePath = process.platform === "linux" ? "/dev/stdin" : "./example.txt";

let input = fs.readFileSync(filePath).toString().split("\n");
let [nm, A, B] = input;
let [n, m] = nm.split(" ");
let i = 0;
let j = 0;
let answer = [];
let arrA = A.split(" ").map((ele) => Number(ele));
let arrB = B.split(" ").map((ele) => Number(ele));

while (i < n && j < m) {
  if (arrA[i] < arrB[j]) {
    answer.push(arrA[i++]);
  } else {
    answer.push(arrB[j++]);
  }
}
if (i < n) {
  answer = answer.concat(arrA.slice(i));
}
if (j < m) {
  answer = answer.concat(arrB.slice(j));
}
console.log(answer.join(" "));
